src/libs/IOUUtils.ts

```diff
@@ -44,23 +44,25 @@ function updateIOUOwnerAndTotal(iouReport: OnyxEntry<Report>, actorAccountID: nu
     // Make a copy so we don't mutate the original object
     const iouReportUpdate: Report = {...iouReport};
 
-    if (iouReportUpdate.total) {
-        if (actorAccountID === iouReport.ownerAccountID) {
-            iouReportUpdate.total += isDeleting ? -amount : amount;
-        } else {
-            iouReportUpdate.total += isDeleting ? amount : -amount;
-        }
-
-        if (iouReportUpdate.total < 0) {
-            // The total sign has changed and hence we need to flip the manager and owner of the report.
-            iouReportUpdate.ownerAccountID = iouReport.managerID;
-            iouReportUpdate.managerID = iouReport.ownerAccountID;
-            iouReportUpdate.total = -iouReportUpdate.total;
-        }
+    // Let us ensure a valid value before updating the total amount.
```

Comment:
@fedirjh 
As shown, I think it may be even more safer if we just reset the amount to 0 for any invalid value as we must calculate the new total. Let me know what you think